conans/model/info.py

```diff
@@ -330,6 +333,32 @@ def clone(self):
         result.settings_target = self.settings_target.copy() if self.settings_target else None
         return result
 
+    def serialize(self):
```

Comment:
IMO, it's confusing to see that `serialize` returns a dict-like Python object and `dumps` a string one. I don't know if it'd be better to change it to `to_dict` or something like that.